src/audiomixcontroller/DefaultAudioMixController.ts

```diff
@@ -56,13 +56,12 @@ export default class DefaultAudioMixController implements AudioMixController {
         const oldSinkId: string = this.audioElement.sinkId;
         if (newSinkId !== oldSinkId) {
           if (this.browserBehavior.hasChromiumWebRTC()) {
-            new AsyncScheduler().start(async () => {
-              const existingStream = await this.audioElement.srcObject;
-              this.audioElement.srcObject = null;
-              // @ts-ignore
-              await this.audioElement.setSinkId(newSinkId);
+            const existingStream = this.audioElement.srcObject;
+            this.audioElement.srcObject = null;
+            // @ts-ignore
+            this.audioElement.setSinkId(newSinkId);
+            new TimeoutScheduler(50).start(() => {
```

Comment:
Using any timeout-related API within a non-async function is error-prone. These are some examples.

This code will throw an error:
```
mixController.bindAudioElement(elementA);
mixController.unbindAudioElement();
```

`elementA` will have `elementB`'s stream, while the user just want to swich the element.
```
mixController.bindAudioElement(elementA);
mixController.bindAudioElement(elementB);
```

---

The `setSink` failure on Chrome should be a common error. I wonder if there's another solution rather than nullifying it. Do you have the Chromium bug link that says `srcObject` should be `null`?

Reply:
It was based out of experimentation and having the following minimal repro as follows:

//create audio
audio = document.createElement('audio');
audio.autoplay = true

//set sinkId to the bluetooth device
await audio.setSinkId(<bt_hex-id>)
console.log('Audio is being played on ' + audio.sinkId);

//set new media stream
 audio.srcObject= new MediaStream()
 
//confirm current sinkId
audio.sinkId
 
//disconnect bluetooth and set the sinkId to the mac device
await audio.setSinkId(<mac_hex-id>)
console.log('Audio is being played on ' + audio.sinkId);

while the audio.srcObject is not null, and when we try to set up the sinkId, we will see a DOMException

Reply:
Agree that we should avoid setTimeout if possible. It seems hacky.

Reply:
Yeah, we had AsyncScheduler before for setting sinkId. Removed that and now in order to achieve a slight delay before the stream is reattached to the srcObject I had to use TimeoutScheduler.
Trying to figure out an alternative approach.